### PR TITLE
Add small sludge trail to pupating zombies

### LINKED ISSUE #49644: we need a lesser sludge trail
# Is your feature request related to a problem? Please describe.
Sludge trails are a feature from giant slugs that I borrowed from pupating zombies. Unfortunately, the sludge trail from pupating zombies is just overboard given their frequency.

# Describe the solution you'd like
We need a "small_sludge_trail" feature that dispenses sludge, but only about once every other tile or so, and only one sludge tile - not a 3x3 area. Apply it to all the pupating zombie types.

# Describe alternatives you've considered
We could use a shorter-lived sludge field as well or instead.
Alternatively the zombie could occasionally barf out a 5x5 sludge spatter around itself every few minutes.

# Additional context
Now that we have more pupating zombies we need this even more.

### PR DESCRIPTION
**Pull request by dgobi**
#### Summary
Features "Added small sludge trail to pupating zombies."

#### Purpose of change
Fixes #49644 by adding a small sludge trail

#### Describe the solution
Adds a small one tile sludge trail with intensity 1 to all pupating zombies which has a 1 in 2 chance of secreting as they move. 

#### Describe alternatives you've considered
A suggested alternative/addition is a shorter lived sludge field, but I didn't want to make ALL sludge fields shorter lived (for slugs also) and adding a whole new field type felt beyond the scope of this. Please do let me know if I got that wrong!

#### Testing
Spawned in all sludge producing monsters to verify their behaviour in debug.

#### Additional context
This is my first PR ever and I am a C++ novice so I'd appreciate any review and/or advice!

### CODE REVIEW

**Comment by LyleSY** [comment]
👋 Pupating zombie creator here. This is exactly the solution I was hoping for, well done.

**Comment by I-am-Erk** [comment]
You'll need to run a `make-astyle` on this to clean it up before we're able to merge it

File: src/monmove.cpp [hunk 1]
```diff
@@ -1755,6 +1755,12 @@ bool monster::move_to( const tripoint &p, bool force, bool step_on_critter,
         }
     }
 
+    if( has_flag (MF_SMALLSLUDGETRAIL ) ) {
```
Comment (anothersimulacrum) [inline src/monmove.cpp:1758]:
```suggestion
    if( has_flag ( MF_SMALLSLUDGETRAIL ) ) {
```

**Comment by dgobi** [comment]
> 
> 
> You'll need to run a `make-astyle` on this to clean it up before we're able to merge it

Sorry, I apparently missed that when I was going over the various "how to contribute pages". I have now done so and it did remove that one erroneous whitespace which apparently caused this all to fail. However I see you've now made this change, do I still need to do anything?

Clearly I'm in way over my head, should probably have waited a long while to jump into this!

**Comment by actual-nh** [comment]
> Clearly I'm in way over my head, should probably have waited a long while to jump into this!

Oh, no, you're doing fine; forgetting to run astyle after a change is something people who've done multiple PRs still do.

**Pull request merged**